src/Propulsion.Feed/FeedSource.fs

```diff
@@ -30,17 +30,17 @@ type FeedSourceBase internal
         finally ingester.Stop() }
     let mutable partitions = Array.empty<struct(Ingestion.Ingester<_> * FeedReader)>
     let dumpStats () = for _i, r in partitions do r.DumpStats()
-    let rec pumpStats () = async {
-        try do! Async.Sleep statsInterval
+    let rec pumpStats ct : Task = task {
+        try do! Task.delay statsInterval ct
         finally dumpStats () // finally is so we do a final write after we are cancelled, which would otherwise stop us after the Async.Sleep
```

Comment:
```suggestion
        finally dumpStats () // finally is so we do a final write after we are cancelled, which would otherwise stop us after the sleep
```